Buildings/Fluid/AirFilters/BaseClasses/FiltrationEfficiency.mo

```diff
@@ -1,37 +1,37 @@
 within Buildings.Fluid.AirFilters.BaseClasses;
 model FiltrationEfficiency
   "Component that calculates the filtration efficiency"
-  parameter Real mCon_nominal(
-    final unit="kg")
-    "Maximum mass of the contaminant can be captured by the filter";
-  parameter Real epsFun[:]
-    "Filtration efficiency curve";
+  parameter Buildings.Fluid.AirFilters.BaseClasses.Data.Generic per
+    "Record with performance data"
+    annotation (Placement(transformation(extent={{20,62},{40,82}})));
   Buildings.Controls.OBC.CDL.Interfaces.RealInput mCon(
     final unit="kg")
     "Mass of the contaminant captured by the filter"
     annotation (Placement(transformation(extent={{-140,-20},{-100,20}})));
-  Buildings.Controls.OBC.CDL.Interfaces.RealOutput y(
-    final unit="1",
-    final min=0,
-    final max=1)
-    "Filtration efficiency"
+  Buildings.Controls.OBC.CDL.Interfaces.RealOutput y[size(per.substanceName, 1)](
+    each final unit="1",
+    each final min=0,
+    each final max=1) "Filtration efficiency"
     annotation (Placement(transformation(extent={{100,-80},{140,-40}})));
   Buildings.Controls.OBC.CDL.Interfaces.RealOutput rat(
-    final unit="1",
-    final min=0,
-    final max=1)
+    each final unit="1",
+    each final min=0,
+    each final max=1)
     "Relative mass of the contaminant captured by the filter"
     annotation (Placement(transformation(extent={{100,40},{140,80}})));
-
```

Comment:
add a protected section to find out the total types of the contaminant

```
protected
  parameter Integer nCon = size(per.substanceName,1) "Total types of contaminant substances";
```
and then
change above `y[size(per.substanceName,1)` to `y[nCon]`, and below the for loop.

Reply:
Change it as suggested